kubernetes-maven-plugin/it/src/it/simple-with-route-flag-true/expected/kubernetes.yml

```diff
@@ -84,7 +84,7 @@ items:
             valueFrom:
               fieldRef:
                 fieldPath: metadata.namespace
-          image: jkube-maven-sample-zero-config:latest
+          image: jkube/jkube-maven-sample-zero-config:latest
```

Comment:
Is this because previously this file was ignored?

Reply:
Actually, earlier this test was running with `jkube.mode=openshift`. With this PR, runtime mode is KUBERNETES for Kubernetes Maven Plugin.
https://github.com/eclipse/jkube/blob/48ccd83fc8ffd81e3d8716b4fa8b43177ae6755d/kubernetes-maven-plugin/it/src/it/simple-with-route-flag-true/invoker.properties#L16

Reply:
So this file was being ignored previously, right? i.e. if we delete it from master/main branch, tests won't break

Reply:
I don't think this file is ignored.

Reply:
If this is the case, was previous behavior was wrong? Why were `kubernetes.yml` files generated if runtime mode was `openshift`?

Reply:
I think `kubernetes.yml` were getting generated because `kubernetes-maven-plugin` goals were being invoked. This `jkube.mode` property was changing default generator image name depending upon runtime mode. Earlier with `jkube.mode` set to Openshift, they were generating image name for Openshift builds. Right now we have same test in openshift and it's image field looks like this:
https://github.com/eclipse/jkube/blob/b2e25f66df900beede673ad4d236f1f8e900eb0e/openshift-maven-plugin/it/src/it/simple-with-route-flag-true/expected/openshift.yml#L90

Reply:
> Right now we have same test in openshift and it's image field looks like this:

Yes, but it's generated in an `openshift.yml` file (as expected).

The fact that this wasn't the case before, and that instead of generating an `openshift.yml` file a `kubernetes.yml` file was being generated with openshift specific configurations (i.e. `image`)  and not even completed with deploymentconfig files, makes me understand that previous behavior was wrong.

So conclusion is that this file was not being ignored before due to an inconsistent behavior, but after the current PR, things will behave as expected and only specific cluster files will be generated with the **right** configs.

Reply:
cool, thanks. I would mark this as resolved.